### PR TITLE
community[patch]: fix Azure AI search delete and improve documentation 

### PR DESCRIPTION
**Pull request by sinedied**
This PRs includes the following changes:
- Fix missing `await initPromise` when doing a delete with filter, causing race conditions
- Set default search mode to hybrid to improve search results relevance
- Add documentation on hybrid search and semantic ranking

### CODE REVIEW

File: libs/langchain-community/src/vectorstores/azure_aisearch.ts [hunk 1]
```diff
@@ -368,7 +369,8 @@ export class AzureAISearchVectorStore extends VectorStore {
     k = 4,
     filter: this["FilterType"] | undefined = undefined
   ): Promise<[Document, number][]> {
-    const searchType = this.options.type ?? AzureAISearchQueryType.Similarity;
+    const searchType =
+      this.options.type ?? AzureAISearchQueryType.SimilarityHybrid;
```
Comment (bracesproul) [inline libs/langchain-community/src/vectorstores/azure_aisearch.ts:373]: What's the reason for this change? Technically this is "breaking" to users in the sense that the functionality will be different.

**Review by bracesproul — COMMENTED** [review]
Lgtm, just had one comment

File: docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx [hunk 2]
```diff
@@ -27,6 +27,17 @@ import EnvVars from "@examples/indexes/vector_stores/azure_aisearch/.env.example
 
 <CodeBlock language="text">{EnvVars}</CodeBlock>
 
+## About hybrid search
+
+Hybrid search is a feature that combines the strengths of full text search and vector search to provide the best ranking performance. It's enabled by default in Azure AI Search vector stores, but you can select a different search query type by setting the `search.type` property when creating the vector store.
+
+You can read more about hybrid search and how it may improve your search results in the [official documentation](https://learn.microsoft.com/azure/search/hybrid-search-overview).
+
+In some scenarios like retrieval-augmented generation, you may want to enable **semantic ranking** in addition to hybrid search to improve the relevance of the search results. You can enable semantic ranking by setting the `search.type` property to `AzureAISearchQueryType.SemanticHybrid` when creating the vector store.
```
Comment (chuwik) [inline docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx:36]: nit: may be worth mentioning 'RAG' next to 'retrieval-augmented generation', as some folks may only be familiar with the acronym.

File: docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx [hunk 3]
```diff
@@ -27,6 +27,17 @@ import EnvVars from "@examples/indexes/vector_stores/azure_aisearch/.env.example
...
+Note that semantic ranking capabilities is only available in the Basic and higher pricing tier, and subject to [regional availability](https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=search).
```
Comment (chuwik) [inline docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx:37]: nit: [...]semantic ranking capabilities **are** only

Comment (chuwik) [inline docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx:37]: nit: [...]and higher pricing **tiers**

File: docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx [hunk 4]
```diff
@@ -27,6 +27,17 @@ import EnvVars from "@examples/indexes/vector_stores/azure_aisearch/.env.example
...
+
+You can read more about using semantic ranking performance with hybrid search in [this blog post](https://techcommunity.microsoft.com/t5/ai-azure-ai-services-blog/azure-cognitive-search-outperforming-vector-search-with-hybrid/ba-p/3929167).
```
Comment (chuwik) [inline docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx:39]: nit: [...]about **the performance of using semantic ranking with hybrid search**

**Review by chuwik — APPROVED** [review]
LGTM, left some nits.

File: libs/langchain-community/src/vectorstores/azure_aisearch.ts (hunk 1 above)
Comment (farzad528) [reply libs/langchain-community/src/vectorstores/azure_aisearch.ts:373]: Hybrid search is the default retreival mode for other vector stores that support it. We'd like to align to that pattern as hybrid proves better retrevial quality vs pure vector search.

**Review by farzad528 — APPROVED** [review]

File: docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx (hunk 4 above)
Comment (farzad528) [reply docs/core_docs/docs/integrations/vectorstores/azure_aisearch.mdx:39]: nit... don't hard code "en-us" in the URL locale

File: libs/langchain-community/src/vectorstores/azure_aisearch.ts (hunk 1 above)
Comment (sinedied) [reply libs/langchain-community/src/vectorstores/azure_aisearch.ts:373]: We had a discussion with the PMs regarding which search mode would bring the best relevance in most scenarios as a default value, and concluded that hybrid search should be the best for most use cases.
This is technically breaking, but people relying on defaults should get slightly improved relevance without any other impact on their code.

**Comment by sinedied** [comment]
@bracesproul We've answered your comment, is there anything else you need from our side?

**Comment by bracesproul** [comment]
> @bracesproul We've answered your comment, is there anything else you need from our side?

Nope, lgtm and will get a release asap!
